scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/TaskLauncher.java

```diff
@@ -449,11 +453,33 @@ public void getStoredLogs(AppenderProvider logSink) {
         taskLogger.getStoredLogs(logSink);
     }
 
-    public PublicKey generatePublicKey() throws NoSuchAlgorithmException {
-        KeyPairGenerator keyGen;
-        keyGen = KeyPairGenerator.getInstance("RSA");
-        keyGen.initialize(1024, new SecureRandom());
-        KeyPair keyPair = keyGen.generateKeyPair();
+    /**
+     * this method exist as a separate method only for test purpose
+     * because in some test (e.g. TaskLauncherTest) we do not mock PAActiveObject
+     * so it is easy to mock this method.
+     * @return
+     * @throws NodeException
+     */
+    public String nodeUrl() throws NodeException {
+        return PAActiveObject.getNode().getNodeInformation().getURL();
+    }
+
+    private static KeyPair getKeyPair(String nodeUrl) throws NodeException, NoSuchAlgorithmException {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the declaration of thrown exception 'org.objectweb.proactive.core.node.NodeException', as it cannot be thrown from method's body. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)